### PR TITLE
Some Xembed fixes

### PR DESCRIPTION
**Pull request by psychon**
Here are two things that I noticed while looking into https://github.com/awesomeWM/awesome/issues/891. Neither helps with the issue described there (I think...), but hopefully conforming to the spec somewhere somewhen helps someone.

As mentioned in the second commit message, this isn't properly tested. Let's just see if things still work...

### CODE REVIEW

**Comment by psychon** [comment]
The previous version broke Skype's tray icon (there are specifications and there is the reality... Skype (Qt?) lives in the reality). This is fixes now.

Adding to next.

File: common/xembed.c [hunk 1]
```diff
@@ -24,6 +24,9 @@
 #include "common/util.h"
 #include "common/atoms.h"
 
+/* I should really include the correct header instead... */
```
Comment (blueyed) [inline common/xembed.c]: Will you do it? :)

File: event.c [hunk 2]
```diff
@@ -724,10 +725,13 @@ event_handle_maprequest(xcb_map_request_event_t *ev)
     if(wa_r->override_redirect)
         goto bailout;
 
-    if(xembed_getbywin(&globalconf.embedded, ev->window))
+    if((em = xembed_getbywin(&globalconf.embedded, ev->window)))
     {
         xcb_map_window(globalconf.connection, ev->window);
         xembed_window_activate(globalconf.connection, ev->window);
+        /* This is not the correct way to set this, but let's be gentle... */
+        em->info.flags |= XEMBED_MAPPED;
```
Comment (blueyed) [inline event.c]: What would be the correct way?

File: common/xembed.c (hunk 1 above)
Comment (psychon) [inline common/xembed.c]: Nope. At least I call myself out for this uglyness. ;-)

The problem is that the stuff in `common/` should be self-contained and this dependency goes against that and i have no good idea what to do about it.

File: event.c (hunk 2 above)
Comment (psychon) [inline event.c]: Actually set an `_XEMBED_INFO` property on the tray icon, like the xembed specification say (which Qt/Skype does not set at all). Neither of the xembed spec nor the systray spec mention anything about MapWindow.

Comment (blueyed) [inline event.c]: Would be nice to have this in the comment with the code then.

**Comment by blueyed** [comment]
:+1: (just quickly tested it locally)

**Comment by psychon** [comment]
Ok, I'll squash the following diff into this when I find the time:
```diff
diff --git a/event.c b/event.c
index 74591b2..c4e2ea3 100644
--- a/event.c
+++ b/event.c
@@ -729,7 +729,11 @@ event_handle_maprequest(xcb_map_request_event_t *ev)
     {
         xcb_map_window(globalconf.connection, ev->window);
         xembed_window_activate(globalconf.connection, ev->window);
-        /* This is not the correct way to set this, but let's be gentle... */
+        /* The correct way to set this is via the _XEMBED_INFO property. Neither
+         * of the XEMBED not the systray spec talk about mapping windows.
+         * Apparently, Qt doesn't care and does not set an _XEMBED_INFO
+         * property. Let's simulate the XEMBED_MAPPED bit.
+         */
         em->info.flags |= XEMBED_MAPPED;
         luaA_systray_invalidate();
     }
```

**Comment by psychon** [comment]
Ok. I updated this and it could now be merged.

**Pull request merged**